plugins/woocommerce/src/Admin/API/OnboardingPlugins.php

```diff
@@ -124,6 +125,29 @@ function ( $value ) {
 				),
 			)
 		);
+
+		/*
+		 * This is a temporary solution to override /jetpack/v4/connection/data endpoint
+		 * registered by Jetpack Connection when Jetpack is not installed.
+		 *
+		 * For more details, see https://github.com/woocommerce/woocommerce/issues/38979
+		 */
+		if ( Constants::get_constant( 'JETPACK__VERSION' ) === null ) {
```

Comment:
To reduce the chance of unknown side-effects, perhaps we could also add an [wp_is_mobile](https://developer.wordpress.org/reference/functions/wp_is_mobile/) check to limit the change to mobile devices only (or even better if we could limit it to requests from the android app).

Reply:
That's a good suggestion! I'll make a change.

> What would be a good way to test the connection flow still works from the package?

There's no change for the existing flows. Just make sure to delete Jetpack before testing.

Reply:
Updated in https://github.com/woocommerce/woocommerce/pull/38987/commits/203722e4c3e6c89e922cc7fe4b79fb55095dc0a6